File: Leetcode 75/Level1/Day 2/Is-Subsequence.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
const isSubsequence = (s, t) => {
  let [sPointer, tPointer] = [0, 0];

  while (sPointer < s.length && tPointer < t.length) {
    if (s[sPointer] === t[tPointer]) {
      sPointer++;
    }
    tPointer++;
  }

  return sPointer >= s.length;
};

File: Go/Is-Subsequence.go
package main

func main() {}

func isSubsequence(s string, t string) bool {
	sPointer, tPointer := 0, 0

	for tPointer < len(t) && sPointer < len(s) && tPointer < len(t) {
		if s[sPointer] == t[tPointer] {
			sPointer++
		}
		tPointer++
	}

	return sPointer >= len(s)
}